src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,88 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= LOAD_FACTOR * table.length) {
+            resize();
+        }
+        int hash = hash(key);
+        Node<K, V> newNode = new Node(hash, key, value, null);
+        Node<K, V> node = table[hash];
+        if (node == null) {
+            table[hash] = newNode;
+        } else {
+            while (node != null) {
+                if (Objects.equals(key, node.key)) {
+                    node.value = value;
+                    return;
+                }
+                if (node.next == null) {
+                    node.next = newNode;
+                    break;
+                }
+                node = node.next;
+            }
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> node = table[hash(key)];
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        size = 0;
+        int newSize = table.length * 2;
+        Node<K, V>[] oldTable = table;
+        table = (Node<K, V>[]) new Node[newSize];
+        for (Node<K, V> node : oldTable) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
+
+    private int hash(K key) {
+        return Math.abs(key == null ? 0 : key.hashCode() % table.length);
+    }
+
+    private static class Node<K, V> {
+        private int hash;
```

Comment:
this field is redundant inside the Node. refactor the class and constructor please